### ISSUE TITLE
Dot products between pairs of ArraySparseIndexedTensors fails

### ISSUE THREAD

**Issue by riedelcastro** [factorie/factorie#131]
I think the culprit is here

``` scala
        val v1 = if (this.__npos < v.__npos) this else v
        val v2 = if (v.__npos < this.__npos) v else this
```

This means either both v1 and v2 are this, or v.

**Comment by riedelcastro** [comment]
The fix
```
        val v1 = if (this.__npos < v.__npos) this else v
        val v2 = if (v.__npos < this.__npos) this else v
```
only works if both vectors have different `npos` values. If they are the same both v1 and v2 are v.

**Comment by strubell** [comment]
Thanks for pointing this out, should be (actually) resolved in PR #302
